src/api/libcellml/issue.h

```diff
@@ -310,121 +283,279 @@ class LIBCELLML_EXPORT Issue
-     *
-     * @return A pointer to the units that this issue was raised on.
-     * If no units has been set for this issue, return a @c nullptr.
+     * @return A @c UnitsPtr relevant to this issue, or
+     *         a @c nullptr if the internal type is not @c ItemType::UNITS.
      */
     UnitsPtr units() const;
 
     /**
      * @brief Set the variable for this issue.
      *
-     * Set the @p variable that this issue is relevant to.
+     * The internal type will be set to @c ItemType::VARIABLE.
      *
-     * @param variable A pointer to the variable this issue is relevant to.
+     * @param variable A @c VariablePtr relevant to this issue.
      */
     void setVariable(const VariablePtr &variable);
 
     /**
-     * @brief Get the variable for this issue.
-     *
-     * Get the variable that this issue is relevant to.
+     * Get the variable relevant to this issue.
      *
-     * @return A pointer to the variable this variable issue was raised on.
-     * If no variable has been set for this issue, return a @c nullptr.
+     * @return A @c VariablePtr relevant to this issue, or
+     *         a @c nullptr if the internal type is not @c ItemType::VARIABLE.
      */
     VariablePtr variable() const;
 
     /**
      * @brief Set the reset for this issue.
      *
-     * Set the @p reset that this issue is relevant to.
+     * The internal type will be set to @c ItemType::RESET.
      *
-     * @param reset A pointer to the reset this issue is relevant to.
+     * @param reset A @c ResetPtr relevant to this issue.
      */
     void setReset(const ResetPtr &reset);
 
     /**
-     * @brief Get the reset for this issue.
+     * Get the reset relevant to this issue.
      *
-     * Get the reset that this issue is relevant to.
-     *
-     * @return A pointer to the reset this reset issue was raised on.
-     * If no reset has been set for this issue, return a @c nullptr.
+     * @return A @c ResetPtr relevant to this issue, or @c nullptr
+     *         if the internal type is not @c ItemType::RESET.
      */
     ResetPtr reset() const;
 
+    /**
+     * @brief Set the component whose MathML is relevant to this issue.
+     *
+     * The internal type will be set to @c ItemType::MATHML.
+     *
+     * @param component The @c ComponentPtr whose MathML is relevant to this issue.
+     */
+    void setMath(const ComponentPtr &component);
+
+    /**
+     * Get the component whose MathML is relevant to this issue.
+     *
+     * @return A @c ComponentPtr whose MathML is relevant to this issue, or @c nullptr
+     *         if the internal type is not @c ItemType::MATHML.
+     */
+    ComponentPtr math() const;
+
+    /**
+     * @brief Set the connection relevant to this issue.
+     *
+     * The internal type will be set to @c ItemType::CONNECTION.
+     *
+     * @param pair The @c VariablePair whose connection is relevant to this issue.
+     */
+    void setConnection(const VariablePair &pair);
```

Comment:
There are a few methods that require an `std::pair` as a parameter (like `setConnection()` indeed) or return an `std::pair`, and I am really not keen on it (and the same elsewhere in the API if we have something similar).

`std::pair` is a pure C++ construct, so I wouldn't expect something using libCellML having to create a pair and pass it to a libCellML method. This is, for instance, why we have all those `create()` methods, so that end-users don't have to "manually" do something like `std::shared_ptr<Issue> {new Issue {}}`.

In the same way, when we retrieve a pair, we have to do something like `x.first` and `x.second`, which is not very meaningful to me. Instead, I would expect to do something like `x.variable1()` and `x.variable2()` (when it comes to a mapping pair, for instance).

Reply:
You think we should wrap a simple class around std::pair?

Reply:
Yes, I think this would make things cleaner from an API viewpoint. Also, when it comes to calling something like `setConnection()`, I think it would be nice to have two signatures:
1. `void setConnection(const VariablePairPtr &variablePair);`; and
2. `void setConnection(const VariablePtr &variable1, const VariablePtr &variable2);`.

Reply:
Is there an issue to track this one going forward? not something we need to worry about in this PR, right?

Reply:
Yes, I guess this shouldn't stop this PR from being merged in. Ok, I am going to create an issue for it.

Reply:
See issue #744.